homeassistant/components/velbus/services.py

```diff
@@ -32,58 +35,76 @@ def setup_services(hass: HomeAssistant) -> None:
     """Register the velbus services."""
 
     def check_entry_id(interface: str) -> str:
+        """Check the config_entry for a specific interface."""
         for config_entry in hass.config_entries.async_entries(DOMAIN):
             if "port" in config_entry.data and config_entry.data["port"] == interface:
                 return config_entry.entry_id
         raise vol.Invalid(
             "The interface provided is not defined as a port in a Velbus integration"
         )
 
-    def get_config_entry(interface: str) -> VelbusConfigEntry | None:
-        for config_entry in hass.config_entries.async_entries(DOMAIN):
-            if "port" in config_entry.data and config_entry.data["port"] == interface:
-                return config_entry
-        return None
+    async def get_config_entry(call: ServiceCall) -> VelbusConfigEntry:
+        """Get the config entry for this service call."""
+        if CONF_CONFIG_ENTRY in call.data:
+            entry_id = call.data[CONF_CONFIG_ENTRY]
+        elif CONF_INTERFACE in call.data:
+            # Deprecated in 2025.2, to remove in 2025.9
+            async_create_issue(
+                hass,
+                DOMAIN,
+                "deprecated_interface_parameter",
+                breaks_in_ha_version="2025.9.0",
+                is_fixable=False,
+                severity=IssueSeverity.WARNING,
+                translation_key="deprecated_interface_parameter",
+            )
+            entry_id = call.data[CONF_INTERFACE]
+        else:
+            entry_id = None
+        if entry_id is None:
+            raise ServiceValidationError(
+                "Either config entry or interface must be provided"
+            )
```

Comment:
I think it would be better if you use `vol.Any` to make sure you at least get one

Reply:
how should this work? not sure where to start